pkgs/by-name/te/tetrio-desktop/tetrio-plus.nix

```diff
@@ -0,0 +1,120 @@
+{ lib
+, stdenv
+, fetchFromGitLab
+, wasm-pack
+, rustc
+, rustPlatform
+, wasm-bindgen-cli
+, binaryen
+
+, yarn
+, fetchYarnDeps
+, fixup_yarn_lock
+, nodejs
+, asar
+
+, tetrio-src
+}:
+
+let
+  version = "unstable";
+
+  wasm-bindgen-82 = wasm-bindgen-cli.override {
+    version = "0.2.82";
+    hash = "sha256-BQ8v3rCLUvyCCdxo5U+NHh30l9Jwvk9Sz8YQv6fa0SU=";
+    cargoHash = "sha256-mP85+qi2KA0GieaBzbrQOBqYxBZNRJipvd2brCRGyOM=";
+  };
+
+  src = fetchFromGitLab {
+    owner = "UniQMG";
+    repo = "tetrio-plus";
+    rev = "b13d476a162a2aec4ca0def29eb7a4ada9ebf523";
+    hash = "sha256-GaVbrCZIMmwjwNbxiPrAAwmTGj/sBjjeLP1kSdBomIo=";
+    fetchSubmodules = true;
+  };
+
+  tpsecore = rustPlatform.buildRustPackage {
+    pname = "tpsecore";
+    inherit version src;
+
+    sourceRoot = "${src.name}/tpsecore";
+
+    nativeBuildInputs = [
+      wasm-pack
+      wasm-bindgen-82
+      binaryen
+      rustc.llvmPackages.lld
+    ];
+
+    cargoHash = "sha256-fTCNHkh+QdY3N4zmFKblK7hs+LLMg9HDnxP1ZkdNevg=";
+
+    buildPhase = ''
+      HOME=`pwd` RUSTFLAGS="-C linker=lld" wasm-pack build --target web --release
+    '';
+
+    installPhase = ''
+      cp -r pkg/ $out
+    '';
+
+    doCheck = false;
+  };
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "tetrio-plus";
+  inherit version src;
+
+  nativeBuildInputs = [
+    yarn
+    fixup_yarn_lock
+    nodejs
+    asar
+  ];
+
+  env.offlineCache = fetchYarnDeps {
+    yarnLock = ./yarn.lock;
+    hash = "sha256-VYUh9y6PRc1OTLELkqCxP89Xbesv3Nu+eUq6fkuoQHE=";
+  };
+
+  configurePhase = ''
+    runHook preConfigure
+
+    asar extract ${tetrio-src}/opt/TETR.IO/resources/app.asar out
+    cd out
+    
+    install -m644 ${./package.json} package.json
+    install -m644 ${./yarn.lock} yarn.lock
+
+    export HOME=$(mktemp -d)
+    yarn config --offline set yarn-offline-mirror $offlineCache
+    fixup_yarn_lock yarn.lock
+    yarn install --offline --frozen-lockfile --ignore-platform --ignore-scripts --no-progress --non-interactive
+    patchShebangs node_modules/
+
+    cd ..
+
+    runHook postConfigure
+  '';
+
+  buildPhase = ''
+    runHook preBuild
+
+    ln -s ${tetrio-src}/opt/TETR.IO/resources/app.asar app.asar
+    node ./scripts/build-electron.js
+
+    cp -r $src out/tetrioplus
+
+    chmod 755 out/tetrioplus/source/lib/
```

Comment:
some jank permissions stuff with how im installing files, same with the install command above

Reply:
I think someone with more experience than me should review this part.